src/lib.rs

```diff
@@ -230,14 +230,20 @@ impl From<String> for EarlyExit {
     }
 }
 
+/// Extract the base cmd from a path
+pub fn cmd<'a>(default: &'a String, path: &'a String) -> &'a str {
```

Comment:
I don't think this function should be part of the public. Could you make this private, and move the test you defined in `tests/lib.rs` into `src/lib.rs`, and add it like:

```
...
#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_cmd_extraction() {
        let expected = "test_cmd";
        let path = format!("/tmp/{}", expected);
        let cmd = argh::cmd(&path, &path);
        assert_eq!(expected, cmd);
    }
}
```